src/test/java/StringManipulationTest.java

```diff
@@ -7,142 +7,192 @@
     }
 
     @Test
     public void testRemoveNthCharacter3() {
-        fail("Not yet implemented");
+        manipulatedString.setString("Holy mama");
+        assertEquals("Hl aa", manipulatedString.removeNthCharacter(2, false));
     }
 
     @Test
     public void testRemoveNthCharacter4() {
-        fail("Not yet implemented");
+        manipulatedString.setString("Holy mama");
+        assertEquals("H l   a a", manipulatedString.removeNthCharacter(2, true));
     }
 
     @Test
     public void testRemoveNthCharacter5() {
-        fail("Not yet implemented");
+        manipulatedString.setString("Holy mama");
+        assertEquals("Hol  ma a", manipulatedString.removeNthCharacter(4, true));
     }
 
     @Test
     public void testRemoveNthCharacter6() {
-        fail("Not yet implemented");
+        manipulatedString.setString("Exception testing1");
+        assertEquals("Don't mind this", manipulatedString.removeNthCharacter(0, true));
     }
 
     @Test
     public void testRemoveNthCharacter7() {
-        fail("Not yet implemented");
+        manipulatedString.setString("Exception testing2");
+        assertEquals("Don't mind this", manipulatedString.removeNthCharacter(19, true));
     }
 
     @Test
     public void testGeSubStrings1() {
-        manipulatedstring.setString("This is my string");
-        String [] sStings = manipulatedstring.getSubStrings(3, 4);
+        manipulatedString.setString("This is my string");
+        String [] sStings = manipulatedString.getSubStrings(3, 4);
 
         assertEquals(sStings[0], "my");
         assertEquals(sStings[1], "string");
     }
 
     @Test
     public void testGeSubStrings2() {
-        fail("Not yet implemented");
+        manipulatedString.setString("This is my string");
+        String [] sStings = manipulatedString.getSubStrings(1, 2);
+
+        assertEquals(sStings[0], "This");
+        assertEquals(sStings[1], "is");
     }
     @Test
     public void testGeSubStrings3() {
-        fail("Not yet implemented");
+        manipulatedString.setString("Why there are so many test cases to create");
+        String [] sStings = manipulatedString.getSubStrings(1, 2);
+
+        assertEquals(sStings[0], "Why");
+        assertEquals(sStings[1], "there");
     }
     @Test
     public void testGeSubStrings4() {
-        fail("Not yet implemented");
+        manipulatedString.setString("Why there are so many test cases to create");
+        String [] sStings = manipulatedString.getSubStrings(1, 3);
+
+        assertEquals(sStings[0], "Why");
+        assertEquals(sStings[1], "there");
+        assertEquals(sStings[2], "are");
     }
     @Test
     public void testGeSubStrings5() {
-        fail("Not yet implemented");
+        manipulatedString.setString("Why there are so many test cases to create");
+        String [] sStings = manipulatedString.getSubStrings(0, 6);
+
```

Comment:
How is this testing if an exception is thrown? What are the mechanics behind this?